apps/go-mod-validator/action.yaml

```diff
@@ -0,0 +1,85 @@
+name: go-mod-validator
+description: "Validates dependencies in go.mod files are tested"
+
+inputs:
+  # general inputs
+  checkout-repo:
+    description: "enable git checkout repo"
+    required: false
+    default: "true"
+  checkout-repo-fetch-depth:
+    description: "number of commits to fetch"
+    required: false
+    default: "0"
+  # inputs
+  github_action_ref:
```

Comment:
We need the code in this ref for us to execute `node dist/index.js` command. Usually this problem doesn't exist as node packages are usually run as `using: node`. But we need to run a composite action as we need to add Go compiler to the environment. 
Another way of solving this problem would be to run `go-mod-validator` as a docker container. That way the image has downloaded all the necessary code we need to run the code. Looks like this is how many other language custom actions are run.
Any other ideas on how to enable us to run `node dist/index.js` in this custom action without having to pass an extra input variable `github_action_ref`?

Reply:
> Any other ideas on how to enable us to run node dist/index.js in this custom action without having to pass an extra input variable github_action_ref?

The easiest way is by making the calling workflow use  `setup-go` before this action, then making this action a nodejs action, and not a composite.

Reply:
That looks like the right solution. Shall I go ahead and implement that or wait for other's opinions?

Reply:
Implemented this change. Also added an example in `README`